src/Sylius/Component/Channel/spec/Factory/ChannelFactorySpec.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+ 
+namespace spec\Sylius\Component\Channel\Factory;
+ 
+use PhpSpec\ObjectBehavior;
+use Sylius\Component\Channel\Model\ChannelInterface;
+
+/**
+ * @author Arkadiusz Krakowiak <arkadiusz.krakowiak@lakion.com>
+ */
+class ChannelFactorySpec extends ObjectBehavior
+{
+    function let()
+    {
+        $this->beConstructedWith('Sylius\Component\Channel\Model\Channel');
+    }
+
+    function it_is_initializable()
+    {
+        $this->shouldHaveType('Sylius\Component\Channel\Factory\ChannelFactory');
+    }
+
+    function it_implements_channel_factory_interface()
+    {
+        $this->shouldImplement('Sylius\Component\Channel\Factory\ChannelFactoryInterface');
+    }
+
+    function it_is_a_factory()
+    {
+        $this->shouldHaveType('Sylius\Component\Resource\Factory\Factory');
+    }
+
+    function it_create_named(ChannelInterface $channel)
```

Comment:
IMO `it_creates_named_channel` or `it_creates_channel_with_name` would be a bit better.